atlasexec/atlas_models.go

```diff
@@ -134,6 +134,10 @@ type (
 		SHA     string `json:"SHA,omitempty"`
 		Canary  bool   `json:"Canary,omitempty"`
 	}
+	// AtlasError is returned when a migration applying attempt failed.
+	AtlasError struct {
+		Report any
+	}
 )
```

Comment:
I think it's better to be more explicit here and create an error for each type. e.g.,
```
type MigrateApplyError struct {
    MigrateApply
}

func (e *MigrateApplyError) Error() string { return e.MigrateApply.Error }

type SchemaApplyError struct {
    SchemaApply
}

func (e *SchemaApplyError) Error() string { return e.SchemaApply.Error }
```